File: src/main/java/com/mario6/paper/common/FileUtils.java
package com.mario6.paper.common;


import java.io.File;

/**
 * 文件操作相关工具类
 *
 * @author Mario Luo
 * @date 2018.08.09 17:19
 */
public class FileUtils {

    /**
     * 生成指定文件的父目录
     */
    public static boolean makeParent(String file) {
        File fileObj = new File(file);
        return makeParent(fileObj);
    }

    /**
     * 生成指定文件的父目录
     */
    public static boolean makeParent(File file) {
        boolean success = true;
        if (!file.exists()) {
            File parentFile = file.getParentFile();
            if (!parentFile.exists()) {
                success = parentFile.mkdirs();
            }
            // 创建失败可能由于权限等问题
            if (!parentFile.exists()) {
                success = false;
            }
        }
        return success;
    }

    /**
     * 删除指定文件夹
     *
     * @param dir
     * @return
     */
    public static int deleteDir(String dir) {
        return deleteDir(new File(dir));
    }

    /**
     * 删除指定文件夹
     *
     * @param dir
     * @return
     */
    public static int deleteDir(File dir) {
        int count = 0;
        if (dir.isDirectory()) {
            String[] children = dir.list();
            for (int i = 0; i < children.length; i++) {
                count += deleteDir(new File(dir, children[i]));
            }
            dir.delete();
        } else {
            boolean delete = dir.delete();
            if (delete) count++;
        }
        return count;
    }

    /**
     * 清空目录, 不删除目录本身
     *
     * @param dir
     * @return
     */
    public static int emptyDir(String dir) {
        return emptyDir(new File(dir));
    }

    /**
     * 清空目录, 不删除目录本身
     *
     * @param dir
     * @return
     */
    private static int emptyDir(File dir) {
        int count = 0;
        if (dir.isDirectory()) {
            String[] children = dir.list();
            for (int i = 0; i < children.length; i++) {
                count += deleteDir(new File(dir, children[i]));
            }
        }
        return count;
    }
}

File: src/main/kotlin/util/FileUtils.kt
package com.github.lkqm.paper.util

import java.io.File

/**
 * 文件操作相关工具类
 *
 * @author Mario Luo
 * @date 2018.08.09 17:19
 */
object FileUtils {
    /**
     * 生成指定文件的父目录
     */
    fun makeParent(file: String?): Boolean {
        val fileObj = File(file)
        return makeParent(fileObj)
    }

    /**
     * 生成指定文件的父目录
     */
    fun makeParent(file: File): Boolean {
        var success = true
        if (!file.exists()) {
            val parentFile = file.parentFile
            if (!parentFile.exists()) {
                success = parentFile.mkdirs()
            }
            // 创建失败可能由于权限等问题
            if (!parentFile.exists()) {
                success = false
            }
        }
        return success
    }

    /**
     * 删除指定文件夹
     *
     * @param dir
     * @return
     */
    fun deleteDir(dir: String?): Int {
        return deleteDir(File(dir))
    }

    /**
     * 删除指定文件夹
     *
     * @param dir
     * @return
     */
    fun deleteDir(dir: File): Int {
        var count = 0
        if (dir.isDirectory) {
            val children = dir.list()
            for (i in children.indices) {
                count += deleteDir(File(dir, children[i]))
            }
            dir.delete()
        } else {
            val delete = dir.delete()
            if (delete) count++
        }
        return count
    }

    /**
     * 清空目录, 不删除目录本身
     *
     * @param dir
     * @return
     */
    fun emptyDir(dir: String?): Int {
        return emptyDir(File(dir))
    }

    /**
     * 清空目录, 不删除目录本身
     *
     * @param dir
     * @return
     */
    private fun emptyDir(dir: File): Int {
        var count = 0
        if (dir.isDirectory) {
            val children = dir.list()
            for (i in children.indices) {
                count += deleteDir(File(dir, children[i]))
            }
        }
        return count
    }
}